Fix warnings

EricPedley opened the pull request:
## Summary
Fixed a bunch of console warnings. After this PR, there shouldn't be any more warnings in the console.
## Test Plan
Click on all the places on the site and ensure that no warnings appear in the console upon doing any action.
## Issues
Closes #199, #163

EricPedley committed:
moved components into the "/" route

EricPedley committed:
added key prop to components in list

EricPedley committed:
removed unused imports

EricPedley committed:
added dependencies to useEffect

EricPedley committed:
changed spacing to spacing(0)
MUI spacing is a function that takes a multiplier on the default spacing and returns a pixel value.

EricPedley committed:
removed margin line entirely
it looks more symmetric without any margin on the close button.

EricPedley committed:
fixed functional components cannot be given refs warning
moved ConditionalWrapper to its own file and made it use ForwardRef

EricPedley committed:
fixed warning and improved logging

EricPedley committed:
fixed typo on _isMounted
also added comment

EricPedley committed:
fixed crash when state.newsItems is nullish
also changed the if statement to not pass if length is nullish.

EricPedley committed:
removed default case body from switch

EricPedley committed:
changed != to !==

EricPedley committed:
changed deprecated theme.spacing.unit to theme.spacing()

EricPedley committed:
added keys to lists of components

EricPedley committed:
increased max event listeners on appstore

ChaseC99 commented on client/src/stores/AppStore.js:
```diff
@@ -217,8 +217,7 @@ class AppStore extends EventEmitter {
                 break;
-            default:
-                console.log(`[Warning] AppStore invalid action type: ${action.type}`);
+            default: //do nothing
```
This works for now, but we may want to refactor so that we aren't using multiple stores for AppStore and RightPaneStore

EricPedley replied:
Opened an issue for this

ChaseC99 reviewed (changes requested):
A lot of great work so far; the console is looking hecka clean.

There are still some errors on the "Added Classes" and "Map" tabs. Could you take a look into these? 
_(loaded class schedule was called `test`)_

![image](https://user-images.githubusercontent.com/29494270/152956160-a33371fa-9e45-4db0-8644-f5bc939ccd18.png)

EricPedley committed:
changed list keys to actual unique values

ChaseC99 commented on client/src/components/App/ConditionalWrapper.js:
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+const ConditionalWrapper = React.forwardRef(({ condition, wrapper, children }, ref) => {
+    return condition ? wrapper(children) : children;
+});
```
This is a good abstraction 👍

EricPedley committed:
added keys to polyline components

EricPedley committed:
Merge branch 'main' of https://github.com/icssc-projects/AntAlmanac

EricPedley committed:
Merge branch 'main' into fix-warnings

EricPedley committed:
increased max event listeners on appstore

EricPedley committed:
changed key to be unique

EricPedley committed:
changed key to be unique

EricPedley committed:
got rid of warnings
change == to ===
change deprecated locale() to updateLocale()
add onView prop to Calendar

EricPedley committed:
increased max listeners to 300

ChaseC99 reviewed (approved):
YAY GREAT WORK 🎉  

This PR ended up being wayyyyyy more work than anticipated lol. Great job powering through all of the warnings/errors to clean up our dev environment. While these code changes might not be as glamours as a new feature, it's just as important - if not more - to the longterm success of the project.

Let's 🚢  it
